File: backend/Turistando/Turistando.WebApi/Models/Reserva.cs
using System;
using System.Collections.Generic;

namespace Turistando.WebApi.Models
{
    public class Reserva
    {
        public Reserva()
        {
            this.ReservaRoteiros = new List<ReservaRoteiro>();
        }

        public int Id { get; set; }
        public int IdTurista { get; set; }
        public int IdGuia { get; set; }
        public DateTime DataCriacao { get; set; }
        public DateTime DataReserva { get; set; }
        public string Horario { get; set; }
        public string Local { get; set; }
        public double ValorTotal { get; set; }
        public double Avaliacao { get; set; }

        public virtual Usuario Turista { get; set; }
        public virtual Usuario Guia { get; set; }
        public virtual ICollection<ReservaRoteiro> ReservaRoteiros { get; set; }
    }
}

File: frontend/turistando-react/src/models/reserva.ts
import {Usuario} from "./usuario";
import {ReservaRoteiro} from "./reserva-roteiro";

export class Reserva 
{
    constructor(idTurista: number, idGuia: number) 
    {
        this.Id = 0;
        this.IdTurista = idTurista;
        this.IdGuia = idGuia;
        this.DataCriacao = new Date();
        this.DataReserva = new Date();
        this.Horario = '';
        this.Local = '';
        this.ValorTotal = 0;
        this.Avaliacao = 0;
        this.ReservaRoteiros = new Array<ReservaRoteiro>();
    }

    public Id: number;
    public IdTurista: number;
    public IdGuia: number;
    public DataCriacao: Date;
    public DataReserva: Date;
    public Horario: string;
    public Local: string;
    public ValorTotal: number;
    public Avaliacao: number;

    public Turista: Usuario;
    public Guia: Usuario;
    public ReservaRoteiros: ReservaRoteiro[];
}
